overlord/snapstate/check_snap_test.go

```diff
@@ -425,6 +426,242 @@ version: 2
 	c.Check(err, ErrorMatches, `cannot replace signed gadget snap with an unasserted one`)
 }
 
+func (s *checkSnapSuite) setupKernelGadgetSnaps(st *state.State) {
+	gadgetInfo := &snap.SideInfo{
+		RealName: "gadget",
+		Revision: snap.R(1),
+		SnapID:   "gadget-id",
+	}
+	snapstate.Set(st, "gadget", &snapstate.SnapState{
+		SnapType: string(snap.TypeGadget),
+		Active:   true,
+		Sequence: []*snap.SideInfo{gadgetInfo},
+		Current:  gadgetInfo.Revision,
+	})
+
+	kernelInfo := &snap.SideInfo{
+		RealName: "kernel",
+		Revision: snap.R(2),
+		SnapID:   "kernel-id",
+	}
+	snapstate.Set(st, "kernel", &snapstate.SnapState{
+		SnapType: string(snap.TypeKernel),
+		Active:   true,
+		Sequence: []*snap.SideInfo{kernelInfo},
+		Current:  kernelInfo.Revision,
+	})
+}
+
+func (s *checkSnapSuite) mockGadgetSnap(c *C, gadgetId string) (restore func()) {
+	const gadgetYaml = `name: gadget
+type: gadget
+version: 2
+`
+
+	infoGadget, err := snap.InfoFromSnapYaml([]byte(gadgetYaml))
+	infoGadget.SnapID = gadgetId
+	c.Assert(err, IsNil)
+
+	var openGadgetSnapFile = func(path string, si *snap.SideInfo) (*snap.Info, snap.Container, error) {
+		return infoGadget, emptyContainer(c), nil
+	}
+
+	return snapstate.MockOpenSnapFile(openGadgetSnapFile)
+}
+
+func (s *checkSnapSuite) mockKernelSnap(c *C, kernelId string) (restore func()) {
+	const kernelYaml = `name: kernel
+type: kernel
+version: 2
+`
+
+	infoKernel, err := snap.InfoFromSnapYaml([]byte(kernelYaml))
+	infoKernel.SnapID = kernelId
+	c.Assert(err, IsNil)
+
+	var openKernelSnapFile = func(path string, si *snap.SideInfo) (*snap.Info, snap.Container, error) {
+		return infoKernel, emptyContainer(c), nil
+	}
+
+	return snapstate.MockOpenSnapFile(openKernelSnapFile)
+}
```

Comment:
nitpick: it seems we could combine this function and `mockGadgetSnap` in a single one, since most of their code is the same.